python/ray/autoscaler/commands.py

```diff
@@ -80,23 +84,219 @@ def request_resources(num_cpus=None, bundles=None):
+    cli_logger.old_style = log_old_style
+    cli_logger.color_mode = log_color
+    cli_logger.verbosity = verbose
+    cli_logger.dump_command_output = verbose == 3 # todo: add a separate flag?
+
+    cli_logger.detect_colors()
+
+    def handle_yaml_error(e):
+        cli_logger.error(
+            "Cluster config invalid.\n"
+            "Failed to load YAML file "+cf.bold("{}"),
+            config_file)
+        cli_logger.newline()
+        with cli_logger.verbatim_error_ctx("PyYAML error:"):
+            cli_logger.error(e)
+        cli_logger.abort()
+
+    try:
+        config = yaml.safe_load(open(config_file).read())
+    except yaml.parser.ParserError as e:
+        handle_yaml_error(e)
+    except yaml.scanner.ScannerError as e:
+        handle_yaml_error(e)
+
+    # todo: validate file_mounts, ssh keys, etc.
+
+    importer = NODE_PROVIDERS.get(config["provider"]["type"])
+    if not importer:
+        cli_logger.abort(
+            "Unknown provider type " + cf.bold("{}") + "\n"
+            "Available providers are: {}",
+            config["provider"]["type"],
+            cli_logger.render_list([
+                k for k in NodeProvider.keys()
+                if NODE_PROVIDERS[k] is not None
+            ]))
+        raise NotImplementedError("Unsupported provider {}".format(
+            config["provider"]))
+
+    cli_logger.success("Cluster configuration valid.\n")
+
+    printed_overrides = False
+    def handle_cli_override(key, override):
+        if override is not None:
+            if key in config:
+                nonlocal printed_overrides
+                printed_overrides = True
+                cli_logger.warning(
+                    "`{}` override provided on the command line.\n"
+                    "  Using " +
+                    cf.bold("{}") +
+                    cf.dimmed(
+                        " [configuration file has " + cf.bold("{}") + "]"),
+                    key, override, config[key])
+            config[key] = override
+
+    handle_cli_override("min_workers", override_min_workers)
+    handle_cli_override("max_workers", override_max_workers)
+    handle_cli_override("cluster_name", override_cluster_name)
+
+    if printed_overrides:
+        cli_logger.newline()
+
+    cli_logger.keyvalue("Cluster", config["cluster_name"])
+
+    # disable the cli_logger here if needed
+    # because it only supports aws
+    if config["provider"]["type"] != "aws":
+        cli_logger.old_style = True
     config = _bootstrap_config(config)
+    if config["provider"]["type"] != "aws":
+        cli_logger.old_style = False
+
+    with cli_logger.group(
+            "{} config",
+            PROVIDER_PRETTY_NAMES.get(
+                config["provider"]["type"], "Unknown provider")):
+
+        if config["provider"]["type"] == "aws":
+            tags = {
+                "default":
+                    cli_logger.info["head_instance_profile_src"] == "default"
```

Comment:
this `cli_logger.info` key-value mechanism seems really strange. 

Why is it needed? Can you achieve what you're trying to do without attaching this to the logger?